packages/astro/src/core/build/generate.ts

```diff
@@ -374,18 +374,19 @@ async function generatePath(
 		const result = await callEndpoint(endpointHandler, env, ctx);
 
 		if (result.type === 'response') {
-			throw new Error(`Returning a Response from an endpoint is not supported in SSG mode.`);
+			throwIfRedirectNotAllowed(result.response, opts.settings.config);
+			// If there's no body, do nothing
+			if (!result.response.body) return;
+			body = await result.response.text();
```

Comment:
I think this should be something like `body = await result.response.arrayBuffer().then(ab => new Uint8Array(ab))`.

Not all response bodies, e.g. images, are valid UTF-8 strings so using `Response.text()` returns unexpected data during build.

Example: returning `new Response(new Uint8Array([0xff]))` from a static file endpoint causes Astro (v1.8.0; Node v18.11.0) to generate a file containing `<ef bf bd>` instead of `<ff>`.

Reply:
Thanks @bmdelacruz! In general it’s safest to create a new issue if you’re seeing buggy behaviour as sometimes older issues and PRs aren’t picked up in the triage process. Would you be able to do that?